ops/internal/deployer/deployer.go

```diff
@@ -0,0 +1,161 @@
+package deployer
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/BurntSushi/toml"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// OpDeployer manages the process of building, setting up, and running the op-deployer
+// binary to generate genesis states for L2 chains.
+type OpDeployer struct {
+	lgr                log.Logger
+	l1ContractsRelease string
+	inputStatePath     string
+	rootDir            string
+	deployerVersion    string
+}
+
+// NewOpDeployer creates a new OpDeployer instance.
+func NewOpDeployer(lgr log.Logger, l1ContractsRelease, stateFilepath, rootDir string) (*OpDeployer, error) {
```

Comment:
Could we infer `l1ContractsRelease` from the `stateFilePath`? As in, unmarshal the state.json as an opaque mapping and then lookup `l1ContractsLocator` property?